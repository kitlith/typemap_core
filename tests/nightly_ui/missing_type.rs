use typemap_core::{typemap, Contains};

fn missing_value<Opts: Contains<u32>>(opts: &Opts) {
    println!("{}", opts.get::<u32>());
}

fn main() {
    let map = typemap!(&str = "Hello, world!");
    missing_value(&map);
}
