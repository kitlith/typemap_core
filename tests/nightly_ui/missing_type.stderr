error[E0277]: the trait bound `TyEnd: Contains<u32>` is not satisfied
 --> tests/nightly_ui/missing_type.rs:9:19
  |
9 |     missing_value(&map);
  |     ------------- ^^^^ the trait `Contains<u32>` is not implemented for `TyEnd`
  |     |
  |     required by a bound introduced by this call
  |
  = help: the following other types implement trait `Contains<T>`:
            <Ty<A, R> as Contains<A>>
            <Ty<A, R> as Contains<B>>
            <&Ty<B, R> as Contains<A>>
            <&mut Ty<B, R> as Contains<A>>
  = note: required for `Ty<&str, TyEnd>` to implement `Contains<u32>`
note: required by a bound in `missing_value`
 --> tests/nightly_ui/missing_type.rs:3:24
  |
3 | fn missing_value<Opts: Contains<u32>>(opts: &Opts) {
  |                        ^^^^^^^^^^^^^ required by this bound in `missing_value`
